'TestKMSIntegration.test_upload_new_blob_w_bucket_cmek_enabled' flakes w/ EC error

From [this Kokoro build failure](https://source.cloud.google.com/results/invocations/27fb932e-3234-4aac-a3a2-3715a0b6e76e/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-storage%2Fpresubmit%2Fpresubmit/log):

```python
________ TestKMSIntegration.test_upload_new_blob_w_bucket_cmek_enabled _________

self = <test_system.TestKMSIntegration testMethod=test_upload_new_blob_w_bucket_cmek_enabled>

    def test_upload_new_blob_w_bucket_cmek_enabled(self):
        blob_name = "test-blob"
        payload = b"DEADBEEF"
        alt_payload = b"NEWDEADBEEF"

        kms_key_name = self._kms_key_name()
        self.bucket.default_kms_key_name = kms_key_name
        self.bucket.patch()
        self.assertEqual(self.bucket.default_kms_key_name, kms_key_name)

        blob = self.bucket.blob(blob_name)
        blob.upload_from_string(payload)
        # We don't know the current version of the key.
>       self.assertTrue(blob.kms_key_name.startswith(kms_key_name))
E       AttributeError: 'NoneType' object has no attribute 'startswith'

tests/system/test_system.py:2029: AttributeError
```

Similar to #190.  In this case, reloading the blob after a failure would likely catch up to the expected state.

--- a/tests/system/test_system.py
+++ b/tests/system/test_system.py
@@ -2013,6 +2013,7 @@ class TestKMSIntegration(TestStorageFiles):
 
         self.assertEqual(dest.download_as_bytes(), source_data)
 
+    @RetryErrors(unittest.TestCase.failureException)
     def test_upload_new_blob_w_bucket_cmek_enabled(self):
         blob_name = "test-blob"
         payload = b"DEADBEEF"

--- a/tests/system/test_system.py
+++ b/tests/system/test_system.py
@@ -2013,7 +2013,6 @@ class TestKMSIntegration(TestStorageFiles):
 
         self.assertEqual(dest.download_as_bytes(), source_data)
 
-    @RetryErrors(unittest.TestCase.failureException)
     def test_upload_new_blob_w_bucket_cmek_enabled(self):
         blob_name = "test-blob"
         payload = b"DEADBEEF"
@@ -2026,6 +2025,7 @@ class TestKMSIntegration(TestStorageFiles):
 
         blob = self.bucket.blob(blob_name)
         blob.upload_from_string(payload)
+        retry_429_harder(blob.reload)()
         # We don't know the current version of the key.
         self.assertTrue(blob.kms_key_name.startswith(kms_key_name))
 
@@ -2035,7 +2035,7 @@ class TestKMSIntegration(TestStorageFiles):
         self.assertEqual(blob.download_as_bytes(), alt_payload)
 
         self.bucket.default_kms_key_name = None
-        self.bucket.patch()
+        retry_429_harder(self.bucket.patch)()
         self.assertIsNone(self.bucket.default_kms_key_name)
 
 

